subsys/cpp/cpp_guard.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019 Erwin Rol <erwin@erwinrol.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <sys/__assert.h>
+
+#if !defined(CONFIG_LIB_CPLUSPLUS)
```

Comment:
Maybe move this check into the CMakeLists.txt as there is not need to compile this file if `CONFIG_LIB_CPLUSPLUS` is set.

Reply:
That was just copied from the other cpp files in that directory, maybe make a separate cleanup commit?